neo/Consensus/ConsensusService.cs

```diff
@@ -82,8 +94,30 @@ private void CheckExpectedView(byte view_number)
 
         private void CheckSignatures()
         {
+            if (!context.State.HasFlag(ConsensusState.CommitSent) &&
+                context.Signatures.Count(p => p != null) >= context.M &&
+                context.TransactionHashes.All(p => context.Transactions.ContainsKey(p)))
+            {
+                // Send my commit
+
+                context.State |= ConsensusState.CommitSent;
+                SignAndRelay(context.MakeCommitAgreement());
+
+                Log($"Commit sent: height={context.BlockIndex} hash={context.CommitHash} state={context.State}");
+            }
+        }
+
+        private void OnCommitAgreement(ConsensusPayload payload, CommitAgreement message)
+        {
+            if (context.State.HasFlag(ConsensusState.BlockSent) ||
+                !context.TryToCommit(payload, message)) return;
+
+            Log($"{nameof(OnCommitAgreement)}: height={payload.BlockIndex} hash={message.BlockHash.ToString()} view={message.ViewNumber} index={payload.ValidatorIndex}");
```

Comment:
Could you please move log() before if()? From the log, it looks like CN send out block when it receives only one commitAgreement message. This is very confusing. 

![commit](https://user-images.githubusercontent.com/34025001/47709833-2b8c6500-dc6c-11e8-94cf-65eaaf4e3108.png)
![commit-2](https://user-images.githubusercontent.com/34025001/47709839-2e875580-dc6c-11e8-831f-53c510b44dff.png)